// typeorm-transactional-cls-hooked/dist/common.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.setHookInContext = exports.getHookInContext = exports.setEntityManagerForConnection = exports.getEntityManagerForConnection = exports.getEntityManagerOrTransactionManager = exports.initializeTransactionalContext = exports.NAMESPACE_NAME = void 0;
var cls_hooked_1 = require("cls-hooked");
var typeorm_1 = require("typeorm");
var DebugLog_1 = require("./DebugLog");
exports.NAMESPACE_NAME = '__typeOrm___cls_hooked_tx_namespace';
var TYPE_ORM_KEY_PREFIX = '__typeOrm__transactionalEntityManager_';
var TYPE_ORM_HOOK_KEY = '__typeOrm__transactionalCommitHooks';
exports.initializeTransactionalContext = function () {
    DebugLog_1.debugLog("Transactional@initializeTransactionalContext");
    return cls_hooked_1.getNamespace(exports.NAMESPACE_NAME) || cls_hooked_1.createNamespace(exports.NAMESPACE_NAME);
};
exports.getEntityManagerOrTransactionManager = function (connectionName, entityManager) {
    var context = cls_hooked_1.getNamespace(exports.NAMESPACE_NAME);
    if (context && context.active) {
        return exports.getEntityManagerForConnection(connectionName, context) || entityManager;
    }
    return entityManager || typeorm_1.getManager(connectionName);
};
exports.getEntityManagerForConnection = function (connectionName, context) {
    return context.get("" + TYPE_ORM_KEY_PREFIX + connectionName);
};
exports.setEntityManagerForConnection = function (connectionName, context, entityManager) { return context.set("" + TYPE_ORM_KEY_PREFIX + connectionName, entityManager); };
exports.getHookInContext = function (context) {
    return context === null || context === void 0 ? void 0 : context.get(TYPE_ORM_HOOK_KEY);
};
exports.setHookInContext = function (context, emitter) {
    return context.set(TYPE_ORM_HOOK_KEY, emitter);
};
